### PR TITLE
Simple Payments: inserting shortcode from the list of existing products

### PR DESCRIPTION
**Pull request by jsnajdr**
- added selection radio inputs to the product list and started maintaining the selection state.
- enable the 'Insert' button iff some product is selected.
- when 'Insert' is clicked, serialize the product data into a shortcode and insert it.

### CODE REVIEW

File: client/components/tinymce/plugins/simple-payments/style.scss [hunk 1]
```diff
@@ -75,7 +75,23 @@
...
 	flex: auto;
 	overflow-y: auto;
 }
+
+.editor-simple-payments-modal__list-item {
+	display: flex;
+	align-items: center;
+}
+
+input[type="radio"].editor-simple-payments-modal__list-radio {
```
Comment (lamosty) [inline client/components/tinymce/plugins/simple-payments/style.scss]: I think we prefer avoiding using element selectors.

File: client/components/tinymce/plugins/simple-payments/dialog/index.jsx [hunk 2]
```diff
@@ -27,27 +26,34 @@ class SimplePaymentsDialog extends Component {
 		isEdit: PropTypes.bool.isRequired,
 		onChangeTabs: PropTypes.func.isRequired,
 		onClose: PropTypes.func.isRequired,
+		onInsert: PropTypes.func.isRequired,
```
Comment (lamosty) [inline client/components/tinymce/plugins/simple-payments/dialog/index.jsx]: When switching to Editor HTML mode, I'm getting React props warnings:

image.

Check `client/post-editor/editor-html-toolbar/index` -- `SimplePaymentsDialog` is also used there and it represents the HTML mode of Editor.

File: client/components/tinymce/plugins/simple-payments/plugin.jsx [hunk 3]
```diff
@@ -9,16 +9,15 @@ import { unmountComponentAtNode } from 'react-dom';
...
 
 const simplePayments = editor => {
 	let node;
 	const store = editor.getParam( 'redux_store' );
 
 	editor.on( 'init', () => {
-		node = editor.getContainer().appendChild(
-			document.createElement( 'div' )
-		);
+		node = editor.getContainer().appendChild( document.createElement( 'div' ) );
```
Comment (lamosty) [inline client/components/tinymce/plugins/simple-payments/plugin.jsx]: I noticed in your PRs that you sometimes update/change the code formatting. I think the previous code formatting was valid and nice -- can we please not change it unnecessarily? :) It produces bigger diffs and is not really needed.

File: client/components/tinymce/plugins/simple-payments/style.scss (hunk 1 above)
Comment (jsnajdr) [inline client/components/tinymce/plugins/simple-payments/style.scss]: I need to include the `input[type="radio"]` bit here to make the selector more specific. Just the class name wouldn't be enough to override the `input[type="radio"]` rules from `forms.scss`.

File: client/components/tinymce/plugins/simple-payments/plugin.jsx (hunk 3 above)
Comment (jsnajdr) [inline client/components/tinymce/plugins/simple-payments/plugin.jsx]: I use Prettier to format code -- and it sometimes changes a perfectly valid and nice formatting into another valid and nice one :)

Would it help if I moved the formatting-only changes to a separate commit?

File: client/components/tinymce/plugins/simple-payments/dialog/index.jsx (hunk 2 above)
Comment (jsnajdr) [inline client/components/tinymce/plugins/simple-payments/dialog/index.jsx]: Oh, I had no idea that the dialog is used somewhere else than `tinymce/plugins` 😦 Thanks!

File: client/components/tinymce/plugins/simple-payments/plugin.jsx (hunk 3 above)
Comment (lamosty) [inline client/components/tinymce/plugins/simple-payments/plugin.jsx]: > Would it help if I moved the formatting-only changes to a separate commit?

Hmm, not sure. Is there a Calypso-wide consensus of using Prettier for code formatting? If yes, we should all use it so the codebase is always looking similar. If it's not, there will be always a problem that a person who uses Prettier will unnecessarily modify a code. Alternative: couldn't you configure Prettier to only format the code you introduced in a PR? That would make complete sense to me.

File: client/components/tinymce/plugins/simple-payments/style.scss (hunk 1 above)
Comment (lamosty) [inline client/components/tinymce/plugins/simple-payments/style.scss]: There must be some other way how to accomplish the same. :) /cc @mtias, our styling guru.

Comment (jsnajdr) [inline client/components/tinymce/plugins/simple-payments/style.scss]: The only other way I know is using `!important` :) And if I grep for `input[type=` in Calypso sources, the same trick is used at many other places to override the default styling with a class.

I'd be very interested in learning if there is some more elegant way.

Comment (lamosty) [inline client/components/tinymce/plugins/simple-payments/style.scss]: I think we could style it like this:
```
.editor-simple-payments-modal__list-item .editor-simple-payments-modal__list-radio { ... the styling ... }
```

**Comment by jsnajdr** [comment]
Pushed an update with these changes:
- rebase to the latest master (the currency defaults patch caused some conflicts)
- added support for inserting Simple Payments also to the HTML editor

I didn't make any changes yet to the CSS rules -- I'm slightly more in favor of the `input[type].class` rule compared to a nested `.parent .radio` selector.

**Pull request merged**